compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2314,7 +2314,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             field,
             base_ty,
             expr.hir_id,
-            true,
+            false, // don't mention or suggest non-accessible methods
```

Comment:
Can you just remove this field? The only other `true` is to suggest `.await`ing on a missing method, but that one also provides sketchy suggestion if the method is private:

```
mod private {
    pub struct Foo;

    pub async fn foo() -> Foo { Foo }

    impl Foo {
        fn private(&self) {}
    }
}

async fn test() {
    private::foo().private();
}

fn main() {}
```

Reply:
Perhaps I am intimidated by finding the correct location in `tests/ui/`.